src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,799 @@
+            this.BaseMap.put("u", SANS_VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put("m", SANS_VOWEL_TOKEN + SANS_VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put("e", DIGIT_TOKEN + SANS_VOWEL_TOKEN + SANS_VOWEL_TOKEN.toUpperCase());
+        } else {
+            this.BaseMap.put("d", DIGIT_TOKEN);
+            this.BaseMap.put("l", VOWEL_TOKEN);
+            this.BaseMap.put("u", VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put("m", VOWEL_TOKEN + VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put("e", DIGIT_TOKEN + VOWEL_TOKEN + VOWEL_TOKEN.toUpperCase());
+        }
+        Logger.info("BaseMap values set to: "+BaseMap);
+    }
+
+        
+    /**
+     * Continuously increments a set of ids until the set is completely filled
+     * with unique ids.
+     *
+     * @param original the original set of ds
+     * @param order determines whether or not the ids will be ordered
+     * @param isAuto determines whether or not the ids are AutoId or CustomId
+     * @param amount the amount of ids to be created.
+     * @return A set of unique ids
+     * @throws SQLException - thrown whenever there is an error with the
+     * database.
+     */
+    private Set<Id> rollIds(Set<Id> original, boolean order, long totalPermutations, long amount)
+            throws SQLException, NotEnoughPermutationsException {
+        System.out.println("inside roll ids");
+
+        // Used to count the number of unique ids. Size methods aren't used because int is returned
+        long uniqueIdCounter = 0;
+
+        /* 
+         Declares and initializes a list that holds unique values. 
+         If order matters, tree set is used.
+         */
+        Set<Id> uniqueList;
+        if (order) {
+            uniqueList = new TreeSet<>();
+        } else {
+            uniqueList = new LinkedHashSet<>(original.size());
+        }
+
+        // iterate through every id 
+        for (Id currentId : original) {
+            // counts the number of times an id has been rejected
+            long counter = 0;
+
+            // continuously increments invalid or non-unique ids
+            while (!DatabaseManager.isValidId(currentId) || uniqueList.contains(currentId)) {
+                /* 
+                 if counter exceeds totalPermutations, then id has iterated through every 
+                 possible permutation. Related format is updated as a quick look-up reference
+                 with the number of ids that were inadvertedly been created using other formats.
+                 NotEnoughPermutationsException is thrown stating remaining number of ids.
+                 */
+                if (counter > totalPermutations) {
+                    long amountTaken = totalPermutations - uniqueIdCounter;
+
+                    System.out.println("throwing exception");
+                    Logger.error("Total number of Permutations Exceeded: Total Permutation COunt="+totalPermutations);

```

Comment:
minor: when using log messages like this, use the placeholder syntax instead, so instead of appending to the string here, you'd have

```java
Logger.error("Total number of Permutations Exceeded: Total Permutation Count={}", totalPermutations);
```

This is faster because if `ERROR` logging is disabled, the JVM doesn't need to do the string append.

Also, "o" in "COunt" should be lowercase